src/NServiceBus.Transport.Msmq/DelayedDelivery/DueDelayedMessagePoller.cs

```diff
@@ -0,0 +1,341 @@
+            IDelayedMessageStore delayedMessageStore,
+            int numberOfRetries,
+            Action<string, Exception, CancellationToken> criticalErrorAction,
+            string timeoutsErrorQueue,
+            Dictionary<string, string> faultMetadata,
+            TransportTransactionMode transportTransactionMode,
+            TimeSpan timeToTriggerFetchCircuitBreaker,
+            TimeSpan timeToTriggerDispatchCircuitBreaker,
+            int maximumRecoveryFailuresPerSecond
+        )
+        {
+            txOption = transportTransactionMode == TransportTransactionMode.TransactionScope
+                ? TransactionScopeOption.Required
+                : TransactionScopeOption.RequiresNew;
+            this.delayedMessageStore = delayedMessageStore;
+            errorQueue = timeoutsErrorQueue;
+            this.faultMetadata = faultMetadata;
+            this.numberOfRetries = numberOfRetries;
+            this.dispatcher = dispatcher;
+            fetchCircuitBreaker = new RepeatedFailuresOverTimeCircuitBreaker("MsmqDelayedMessageFetch", timeToTriggerFetchCircuitBreaker,
+                ex => criticalErrorAction("Failed to fetch due delayed messages from the storage", ex, CancellationToken.None));
+
+            dispatchCircuitBreaker = new RepeatedFailuresOverTimeCircuitBreaker("MsmqDelayedMessageDispatch", timeToTriggerDispatchCircuitBreaker,
+                ex => criticalErrorAction("Failed to dispatch delayed messages to destination", ex, CancellationToken.None));
+
+            failureHandlingCircuitBreaker = new FailureRateCircuitBreaker("MsmqDelayedMessageFailureHandling", maximumRecoveryFailuresPerSecond, ex => criticalErrorAction("Failed to execute error handling for delayed message forwarding", ex, CancellationToken.None));
+
+            signalQueue = Channel.CreateBounded<bool>(1);
+            taskQueue = Channel.CreateBounded<Task>(2);
+        }
+
+        public void Start()
+        {
+            tokenSource = new CancellationTokenSource();
+            loopTask = Task.Run(() => Loop(tokenSource.Token));
+            completionTask = Task.Run(() => AwaitHandleTasks());
+        }
+
+        public async Task Stop(CancellationToken cancellationToken = default)
+        {
+            tokenSource.Cancel();
+            await loopTask.ConfigureAwait(false);
+            await completionTask.ConfigureAwait(false);
+        }
+
+        public void Signal(DateTimeOffset timeoutTime)
+        {
+            //If the next timeout is within a minute from now, trigger the poller
+            if (DateTimeOffset.UtcNow.Add(MaxSleepDuration) > timeoutTime)
+            {
+                //If there is something already in the queue we are fine.
+                signalQueue.Writer.TryWrite(true);
+            }
+        }
+
+        /// <summary>
+        /// This method does not accept a cancellation token because it need to finish all the tasks that have been started even if cancellation is under way
```

Comment:
```suggestion
        /// This method does not accept a cancellation token because it needs to finish all the tasks that have been started even if cancellation is under way.
```